statsmodels/regression/process_regression.py

```diff
@@ -0,0 +1,840 @@
+        except Exception:
+            cov_params = None
+
+        class rslt:
+            pass
+
+        r = rslt()
+        r.params = f.x
+        r.normalized_cov_params = cov_params
+        r.optim_retvals = f
+        r.scale = 1
+
+        rslt = ProcessMLEResults(self, r)
+
+        return rslt
+
+    def covariance(self, time, scale_params, smooth_params, scale_data,
+                   smooth_data):
+        """
+        Returns a Gaussian process covariance matrix.
+
+        Parameters
+        ----------
+        time : array-like
+            The time points at which the fitted covariance matrix is
+            calculated.
+        scale_params : array-like
+            The regression parameters for the scaling part
+            of the covariance structure.
+        smooth_params : array-like
+            The regression parameters for the smoothing part
+            of the covariance structure.
+        scale_data : Dataframe
+            The data used to determine the scale parameter,
+            must have len(time) rows.
+        smooth_data: Dataframe
+            The data used to determine the smoothness parameter,
+            must have len(time) rows.
+
+        Returns
+        -------
+        A covariance matrix.
+
+        Notes
+        -----
+        If the model was fit using formulas, `scale` and `smooth` should
+        be Dataframes, containing all variables that were present in the
+        respective scaling and smoothing formulas used to fit the model.
+        Otherwise, `scale` and `smooth` should contain data arrays whose
+        columns align with the fitted scaling and smoothing parameters.
+
+        The covariance is only for the Gaussian process and does not include
+        the white noise variance.
+        """
+
+        if not hasattr(self.data, "scale_design_info"):
+            sca = np.dot(scale_data, scale_params)
+            smo = np.dot(smooth_data, smooth_params)
+        else:
+            sc = patsy.dmatrix(self.data.scale_design_info, scale_data)
+            sm = patsy.dmatrix(self.data.smooth_design_info, smooth_data)
+            sca = np.exp(np.dot(sc, scale_params))
+            smo = np.exp(np.dot(sm, smooth_params))
+
+        return self.cov.get_cov(time, sca, smo)
+
+    def predict(self, params, exog=None, *args, **kwargs):
+        """
+        Obtain predictions of the mean structure.
+
+        Parameters
+        ----------
+        params : array-like
+            The model parameters, may be truncated to include only mean
+            parameters.
+        exog : array-like
+            The design matrix for the mean structure.  If not provided,
+            the model's design matrix is used.
+        """
+
+        if exog is None:
+            exog = self.exog
+        elif hasattr(self.data, "design_info"):
+            # Run the provided data through the formula if present
+            exog = patsy.dmatrix(self.data.design_info, exog)
```

Comment:
this should be handled by the Results predict method.
i.e. it is inherited from base.Results